### PR TITLE
Card detail branch

### PR DESCRIPTION
**Pull request by Krystof-Bum**

### CODE REVIEW

File: flag-app/flag-app-project/src/hooks/useApiData.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import apiClient from "../services/api-client";
+import { CanceledError } from "axios";
+import { Country } from "../interfaces";
+
+export const useApiData = (searchedQuery?: string) => {
+  const [countries, setCountries] = useState<Country[]>([]);
+  const [error, setError] = useState("");
+  const [isLoading, setLoading] = useState(false);
+
+  useEffect(() => {
```
Comment (sdee3) [inline flag-app/flag-app-project/src/hooks/useApiData.ts]: This is bad practise. I strongly recommend you check this out and implement this using React Query:

https://www.youtube.com/watch?v=vxkbf5QMA2g

File: flag-app/flag-app-project/src/hooks/useApiData.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react";
+
+import apiClient from "../services/api-client";
+import { CanceledError } from "axios";
```
Comment (sdee3) [inline flag-app/flag-app-project/src/hooks/useApiData.ts]: I would suggest sorting the imports slightly better:

1. Libraries go on top
2. Your interfaces and components go second
3. CSS imports

Each group would be separated by an empty line, making this import set look better:

´´´
import { useEffect, useState } from "react";
import { CanceledError } from "axios";

import apiClient from "../services/api-client";
import { Country } from "../interfaces";
´´´

Also try to export apiClient in the services/index.ts

File: flag-app/flag-app-project/src/context/CountriesContext.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { Country } from "../interfaces";
```
Comment (sdee3) [inline flag-app/flag-app-project/src/context/CountriesContext.tsx:2]: When you have only 2 imports like here, you don't have to separate them with empty lines

File: flag-app/flag-app-project/src/components/CardDetail.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  const currency = currencies[countryCurrencyCode].name;
+
+  const languageCode = Object.keys(languages)[0];
+  const language = languages[languageCode];
+
+  return (
+    <div>
+      <div
+        onClick={handleClickBack}
+        className={`inline-flex items-center gap-2 border rounded-md shadow-md mb-20 cursor-pointer px-6 py-1 ${
```
Comment (sdee3) [inline flag-app/flag-app-project/src/components/CardDetail.tsx]: This is kind of hard to read, so what I would suggest is:

1. Move all "fixed" class names into a single const: `const baseClassName = inline-flex items-center gap-2 border rounded-md shadow-md mb-20 cursor-pointer px-6 py-1`
2. Also move the mode: `const modeClassName = mode === 'dark' ? 'shadow-white' : ''`
3. Change this className to `{$baseClassName} {$modeClassName}`

File: flag-app/flag-app-project/src/components/Card.tsx [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
-const Card = ({ country }: Props) => {
+const Card = ({ country, handleCardClick }: Props) => {
   const { mode } = useDarkMode();
+
   return (
     <div
-      className={`w-64 rounded-md shadow-md overflow-hidden ${
+      onClick={() => handleCardClick(country)}
+      className={`w-64 rounded-md shadow-md overflow-hidden cursor-pointer ${
         mode === "dark" ? "shadow-white" : ""
```
Comment (sdee3) [inline flag-app/flag-app-project/src/components/Card.tsx]: You can do the same here as for the previous file

File: flag-app/flag-app-project/src/components/BorderCountry.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  const { borders } = country;
+
+  if (!borders) return <p>Country has no borders</p>;
+
+  return (
+    <>
+      {borders.map((border) => (
+        <div
+          className={`border shadow-md px-6 ${
+            mode === "dark" ? "shadow-white" : ""
```
Comment (sdee3) [inline flag-app/flag-app-project/src/components/BorderCountry.tsx]: And here - even though it's not so long, it's good to (at least) establish a variable for something like a "base class name"

**Review by sdee3 — CHANGES_REQUESTED** [review]

File: flag-app/flag-app-project/src/components/CardDetail.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  const countryNameCode = Object.keys(name.nativeName)[0];
+  const nativeName = name.nativeName[countryNameCode].official;
+
+  const domain = tld[0];
+
+  const countryCurrencyCode = Object.keys(currencies)[0];
+  const currency = currencies[countryCurrencyCode].name;
+
+  const languageCode = Object.keys(languages)[0];
+  const language = languages[languageCode];
```
Comment (sdee3) [inline flag-app/flag-app-project/src/components/CardDetail.tsx]: You could try and write a custom hook, `useCountryDetails(country)`, which would return all this data from one "central" place. This adds reusability to your codebase

File: flag-app/flag-app-project/src/components/CardWrapper.tsx [hunk 8]
```diff
@@ -1,21 +1,27 @@
...
 
-  if (isLoading) return <p>Loading...</p>;
-  if (error) return <ErrorMessage searchInput={searchInput} />;
+  if (error) return <ErrorMessage />;
 
   return (
     <div className="cards-layout">
-      {countries.map((country) => (
-        <Card country={country} key={country.cca3} />
+      {countries?.map((country) => (
```
Comment (sdee3) [inline flag-app/flag-app-project/src/components/CardWrapper.tsx]: Do an early return if there are no `countries` above L15, just like for `isLoading` or `error`.

**Review by sdee3 — CHANGES_REQUESTED** [review]

File: flag-app/flag-app-project/src/components/CardWrapper.tsx [hunk 9]
```diff
@@ -12,10 +12,11 @@ export const CardWrapper = ({ handleCardClick }: Props) => {
   const { countries, error, isLoading } = useCountries();
 
   if (error) return <ErrorMessage />;
+  if (!countries) return;
```
Comment (sdee3) [inline flag-app/flag-app-project/src/components/CardWrapper.tsx]: either `return null` or return an actual component - ideally a sentence that says "No countries available or found". Returning this, which is basically `undefined`, is unfortunately not good

**Review by sdee3 — CHANGES_REQUESTED** [review]

File: flag-app/flag-app-project/src/components/CardWrapper.tsx (hunk 9 above)
Comment (Krystof-Bum) [reply flag-app/flag-app-project/src/components/CardWrapper.tsx]: yes, good idea

**Review by sdee3 — APPROVED** [review]

**Pull request merged**